util/status.go

```diff
@@ -0,0 +1,44 @@
+package util
+
+import (
+	"net"
+	"sort"
+
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+// StatusUnchanged to check if the new ingress state matches the existing state.
+func StatusUnchanged(existing, new []v1.LoadBalancerIngress) bool {
+	if len(existing) != len(new) {
+		return false
+	}
+	for i, loadbalancer := range existing {
+		if loadbalancer != new[i] {
+			return false
+		}
+	}
+	return true
+}
+
+// SliceToStatus to convert a slice of strings to ingress loadbalancer objects.
+// Allows hostnames or ip addresses and sets the appropriate field.
+func SliceToStatus(endpoints []string) []v1.LoadBalancerIngress {
+	lbi := []v1.LoadBalancerIngress{}
+	for _, ep := range endpoints {
+		if net.ParseIP(ep) == nil {
```

Comment:
perhaps remove the `== nil`, and swap the outcomes around to improve readability